### PR TITLE
refactor: improve performance of checking for offline deployments

### PR DESCRIPTION
**Pull request by MohamedElmdary**
### Description
checking for offline deployments was loading all nodes on grid

### Changes
Load only nodes which used in my contracts

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1373

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/playground/src/components/tf_notification.vue [hunk 1]
```diff
@@ -1,77 +1,78 @@
...
-  const myContracts: GqlContracts = await grid!.contracts.listMyContracts();
-  const contracts: (GqlNodeContract | GqlRentContract)[] = [...myContracts.nodeContracts, ...myContracts.rentContracts];
+    const ids = uniq(contracts.map(c => c.nodeID));
+    const settledNodes = await Promise.allSettled(ids.map(id => gridProxyClient.nodes.byId(id)));
 
-  const userOfflineDeployments = [];
-  const withPubIp = [];
+    const fakeNode = { status: "up" } as GridNode;
+    const offlineAndStandbyNodes = settledNodes
+      .map(n => (n.status === "fulfilled" ? n.value : fakeNode))
```
Comment (0oM4R) [inline packages/playground/src/components/tf_notification.vue:27]: Why do we set status:up here? Can you explain this, please?

Comment (MohamedElmdary) [reply packages/playground/src/components/tf_notification.vue:27]: to filter it

**Review by amiraabouhadid — APPROVED** [review]
verified 
image

File: packages/playground/src/components/tf_notification.vue [hunk 2]
```diff
@@ -1,77 +1,78 @@
...
 
-async function checkOfflineDeployments(grid: GridClient | null) {
-  const offlineNodesids = (await getAllNodes(grid, { flat: true, status: NodeStatus.down })) as number[];
-  const standByNodesids = (await getAllNodes(grid, { flat: true, status: NodeStatus.standBy })) as number[];
-  const offlineAndStandbyNodes = [...offlineNodesids, ...standByNodesids];
+async function checkOfflineDeployments() {
+  try {
+    const myContracts: GqlContracts = await gridStore.client.contracts.listMyContracts();
+    type Contract = GqlNodeContract | GqlRentContract;
+    const contracts: Contract[] = [...myContracts.nodeContracts, ...myContracts.rentContracts];
```
Comment (zaelgohary) [inline packages/playground/src/components/tf_notification.vue:20]: Why isn't name contracts included?

Comment (MohamedElmdary) [reply packages/playground/src/components/tf_notification.vue:20]: `nameContracts` don't have NodeID

**Comment by zaelgohary** [comment]
Great fix ya @MohamedElmdary!

File: packages/playground/src/components/tf_notification.vue [hunk 3]
```diff
@@ -1,77 +1,78 @@
...
-  const contracts: (GqlNodeContract | GqlRentContract)[] = [...myContracts.nodeContracts, ...myContracts.rentContracts];
+    const ids = uniq(contracts.map(c => c.nodeID));
+    const settledNodes = await Promise.allSettled(ids.map(id => gridProxyClient.nodes.byId(id)));
 
-  const userOfflineDeployments = [];
-  const withPubIp = [];
+    const fakeNode = { status: "up" } as GridNode;
+    const offlineAndStandbyNodes = settledNodes
+      .map(n => (n.status === "fulfilled" ? n.value : fakeNode))
+      .filter(n => n.status !== "up")
```
Comment (AhmedHanafy725) [inline packages/playground/src/components/tf_notification.vue:28]: can't we make the first map and filter in one step? and remove the `fakeNode`

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**